### PR TITLE
GH-767 Adds selection to C3 charts

### PR DESCRIPTION
**Pull request by jbrundage**
Fixes GH-767

@mlzummo @dtsnell4 Please review

http://rawgit.com/jbrundage/Visualization/GH-767-AddC3Sel/demos/dermatology.html?src/c3chart/Bar

Signed-off-by: Jay Brundage <jaman.brundage@lexisnexis.com>

### CODE REVIEW

File: src/c3chart/CommonND.js [hunk 1]
```diff
@@ -16,7 +16,7 @@
         };
 
         this._config.data.onclick = function (d, element) {
-            context.click(context.rowToObj(context._data[d.index]), d.id);
+            context.click(context.rowToObj(context.data()[d.index]), d.id);
```
Comment (mlzummo) [inline src/c3chart/CommonND.js]: need to add this -> https://github.com/hpcc-systems/Visualization/blob/master/src/chart/Column.js#L89

Comment (mlzummo) [inline src/c3chart/CommonND.js]: https://github.com/hpcc-systems/Visualization/blob/master/src/amchart/CommonSerial.js#L414

**Comment by mlzummo** [comment]
also need to add to common2d as well?

File: src/c3chart/CommonND.js (hunk 1 above)
Comment (GordonSmith) [inline src/c3chart/CommonND.js]: You need to pass in a third param which is true/false depending on if the click selects or de-selects (which is what Matt is pointing out)

Comment (jbrundage) [inline src/c3chart/CommonND.js]: Will do

**Comment by jbrundage** [comment]
@mlzummo the Common1D, Common2D and CommonND files extend Common

**Comment by mlzummo** [comment]
@jbrundage this needs to be updated as well: 
https://github.com/hpcc-systems/Visualization/blob/candidate-1.8.2/src/c3chart/Common2D.js#L19

File: src/c3chart/Common.js [hunk 2]
```diff
@@ -21,6 +22,10 @@
             data: {}
         };
         this._prevColumnIDs = [];
+        
+        this._config.data.onclick = function (d, element) {
+            context.click(context.rowToObj(context.data()[d.index]), d.id, context.c3Chart.selected().length);
```
Comment (GordonSmith) [inline src/c3chart/Common.js]: I would prefer if ```context.c3Chart.selected().length``` was ```context.c3Chart.selected().length > 0``` to force the boolean.

File: src/c3chart/Common.js [hunk 3]
```diff
@@ -21,6 +22,10 @@
             data: {}
         };
         this._prevColumnIDs = [];
+        
+        this._config.data.onclick = function (d, element) {
+            context.click(context.rowToObj(context.data()[d.index]), d.id, context.c3Chart.selected().length > 0);
```
Comment (GordonSmith) [inline src/c3chart/Common.js]: The click event should be (row, col, selected) you are now omitting the col info...

Comment (jbrundage) [inline src/c3chart/Common.js]: **d.id** is the column name

Comment (GordonSmith) [inline src/c3chart/Common.js]: Whoops sorry (putting glasses back on)...

**Comment by GordonSmith** [comment]
Pie chart is broke (but I am not sure if that is due to these changes):  http://rawgit.com/jbrundage/Visualization/GH-767-AddC3Sel/demos/dermatology.html?src/c3chart/Pie

**Pull request merged**